batch-indexing/src/main/java/data/util/IndexingSettingsGenerator.java

```diff
@@ -0,0 +1,105 @@
+package data.util;
+
+import eu.europeana.indexing.IndexingSettings;
+import eu.europeana.indexing.exception.IndexingException;
+import java.net.InetSocketAddress;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class IndexingSettingsGenerator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(IndexingSettingsGenerator.class);
+
+    protected final IndexingProperties properties;
+
+    public IndexingSettingsGenerator(IndexingProperties properties) {
+        this.properties = properties;
+
+    }
+    public IndexingSettings generate() throws IndexingException, URISyntaxException {
+        if (isNotDefinedFor(properties)) {
+            return null;
+        }
+        IndexingSettings indexingSettings = new IndexingSettings();
+        prepareSettingFor(properties, indexingSettings);
+        return indexingSettings;
+    }
+
+    private boolean isNotDefinedFor(IndexingProperties indexingProperties) {
```

Comment:
isNotDefinedFor? What is "For" here?
Same for the other "For" suffices in this file

Reply:
in general it means `for the object provided in the method parameter`. 
For example: `prepareSettingFor` means 'prepare settings (IndexingSettings instance) for the provided properties (IndexingProperties instance)'.

This class was copy-pasted from Indexing topology. I didn't pay much attention to its structure and correctness.

Reply:
It is not very clear to me personally, but I understand.
Let's leave it as such and a future cleanup could take care of that.